src/item/Armor.php

```diff
@@ -160,12 +184,25 @@ protected function deserializeCompoundTag(CompoundTag $tag) : void{
 		}else{
 			$this->customColor = null;
 		}
+		$trimTag = $tag->getTag(self::TAG_TRIM);
+		if($trimTag instanceof CompoundTag){
+			$material = ArmorTrimMaterial::tryFrom($trimTag->getString(self::TAG_TRIM_MATERIAL, ""));
+			$pattern = ArmorTrimPattern::tryFrom($trimTag->getString(self::TAG_TRIM_PATTERN, ""));
+			if($material instanceof ArmorTrimMaterial && $pattern instanceof ArmorTrimPattern){
+				$this->armorTrim = new ArmorTrim($material, $pattern);
+			}
+		}
 	}
 
 	protected function serializeCompoundTag(CompoundTag $tag) : void{
 		parent::serializeCompoundTag($tag);
 		$this->customColor !== null ?
 			$tag->setInt(self::TAG_CUSTOM_COLOR, Binary::signInt($this->customColor->toARGB())) :
 			$tag->removeTag(self::TAG_CUSTOM_COLOR);
+		$this->armorTrim !== null ?
+			$tag->setTag(self::TAG_TRIM, CompoundTag::create()
+				->setString(self::TAG_TRIM_MATERIAL, $this->armorTrim->getMaterial()->value)
```

Comment:
The save data values shouldn't be directly linked to the enums. We may need to save multiple different types of save data.
A mapping in `pocketmine\data\bedrock` should be used instead.

Reply:
I tried it, but I don't understand where the translation between the save-id to the real name happens...Trims are also invisible after reconnect, but I'm not able to add the same trim on one item again. I will take a few more looks into this, otherwise I will commit everything for you to fix it